jar-tool/Main.hx

```diff
@@ -0,0 +1,108 @@
+package;
+
+import haxe.io.Path;
+import sys.FileSystem;
+import java.StdTypes.Int8;
+import java.NativeArray;
+import java.io.FileOutputStream;
+import java.io.File;
+import java.nio.file.Files;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes.Opcodes_Statics;
+import java.Lib.println;
+
+using StringTools;
+
+/**
+ * HOW TO USE:
+ *
+ * 1) Unzip JAR file into some folder. For example "jar_content";
+ * 2) Run `java -jar Main.jar jar_content`;
+ * 3) Pick up generated `hxjava-std.jar`.
+ */
+final class Main {
+	static function main():Void {
+		function process(dir:String) {
+			if (FileSystem.exists(dir)) {
+				for (file in FileSystem.readDirectory(dir)) {
+					final path = Path.join([dir, file]);
+					if (!FileSystem.isDirectory(path)) {
+						if (path.endsWith(".class")) {
+							println('Process $path... ');
+							final obj:NativeArray<Int8> = Files.readAllBytes(new File(path).toPath());
+							final reader = new ClassReader(obj);
+							final writer = new ClassWriter(ClassWriter.COMPUTE_FRAMES | ClassWriter.COMPUTE_MAXS);
```

Comment:
I just noticed the `COMPUTE_FRAMES` bit. I suppose that means it already re-calculates stack frames, so we don't have to do anything by hand.